File: app/src/main/java/com/arejas/dashboardofthings/utils/Enumerators.java
package com.arejas.dashboardofthings.utils;

public class Enumerators {

    public enum ElementType {
        NETWORK,
        SENSOR,
        ACTUATOR;

        public static ElementType valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return NETWORK;
                case 1:
                    return SENSOR;
                default:
                    return ACTUATOR;
            }
        }
    }

    public enum NetworkType {
        HTTP,
        MQTT;

        public static NetworkType valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return HTTP;
                default:
                    return MQTT;
            }
        }
    }

    public enum HttpAuthenticationType {
        NONE,
        BASIC;

        public static HttpAuthenticationType valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return NONE;
                default:
                    return BASIC;
            }
        }
    }

    public enum MqttAuthenticationType {
        NONE,
        BASIC;

        public static MqttAuthenticationType valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return NONE;
                default:
                    return BASIC;
            }
        }
    }

    public enum DataType {
        BOOLEAN,
        INTEGER,
        DECIMAL,
        STRING;

        public static DataType valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return BOOLEAN;
                case 1:
                    return INTEGER;
                case 2:
                    return DECIMAL;
                default:
                    return STRING;
            }
        }
    }

    public enum MessageType {
        XML,
        JSON,
        RAW;

        public static MessageType valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return XML;
                case 1:
                    return JSON;
                default:
                    return RAW;
            }
        }
    }

    public enum MqttQosLevel {
        QOS_0,
        QOS_1,
        QOS_2;

        public static MqttQosLevel valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return QOS_0;
                case 1:
                    return QOS_1;
                default:
                    return QOS_2;
            }
        }
    }

    public enum HttpMethod {
        GET,
        PUT,
        POST,
        PATCH;

        public static HttpMethod valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return GET;
                case 1:
                    return PUT;
                case 2:
                    return POST;
                default:
                    return PATCH;
            }
        }

        @Override
        public String toString() {
            switch (ordinal()) {
                case 0:
                    return "GET";
                case 1:
                    return "PUT";
                case 2:
                    return "POST";
                default:
                    return "PATCH";
            }
        }
    }

    public enum LogLevel {
        INFO,
        WARN,
        ERROR,
        NOTIF_NONE,
        NOTIF_WARN,
        NOTIF_CRITICAL;

        public static LogLevel valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return INFO;
                case 1:
                    return WARN;
                case 2:
                    return ERROR;
                case 3:
                    return NOTIF_NONE;
                case 4:
                    return NOTIF_WARN;
                default:
                    return NOTIF_CRITICAL;
            }
        }

        @Override
        public String toString() {
            switch (ordinal()) {
                case 0:
                    return "INFO";
                case 1:
                    return "WARNING";
                case 2:
                    return "ERROR";
                case 3:
                    return "NOTIF_NONE";
                case 4:
                    return "NOTIF_WARN";
                default:
                    return "NOTIF_CRITICAL";
            }
        }
    }

    public enum NotificationType {
        NONE,
        WARNING,
        CRITICAL;

        public static NotificationType valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return NONE;
                case 1:
                    return WARNING;
                default:
                    return CRITICAL;
            }
        }

        @Override
        public String toString() {
            switch (ordinal()) {
                case 0:
                    return "INFO";
                case 1:
                    return "NOTIF_WARN";
                default:
                    return "NOTIF_CRITICAL";
            }
        }
    }

    public enum ElementManagementFunction {
        CREATE,
        UPDATE,
        DELETE;

        public static ElementManagementFunction valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return CREATE;
                case 1:
                    return UPDATE;
                default:
                    return DELETE;
            }
        }

        @Override
        public String toString() {
            switch (ordinal()) {
                case 0:
                    return "CREATE";
                case 1:
                    return "UPDATE";
                default:
                    return "DELETE";
            }
        }
    }

    public enum NotificationState {
        NONE,
        WARNING,
        CRITICAL;

        public static NotificationState valueOf(int ordinal) {
            switch (ordinal) {
                case 0:
                    return NONE;
                case 1:
                    return WARNING;
                default:
                    return CRITICAL;
            }
        }

        @Override
        public String toString() {
            switch (ordinal()) {
                case 0:
                    return "NONE";
                case 1:
                    return "WARNING";
                default:
                    return "CRITICAL";
            }
        }
    }

}

File: app/src/main/java/com/arejas/dashboardofthings/utils/Enumerators.kt
package com.arejas.dashboardofthings.utils

class Enumerators {

    enum class ElementType {
        NETWORK,
        SENSOR,
        ACTUATOR;


        companion object {

            fun valueOf(ordinal: Int): ElementType {
                when (ordinal) {
                    0 -> return NETWORK
                    1 -> return SENSOR
                    else -> return ACTUATOR
                }
            }
        }
    }

    enum class NetworkType {
        HTTP,
        MQTT;


        companion object {

            fun valueOf(ordinal: Int): NetworkType {
                when (ordinal) {
                    0 -> return HTTP
                    else -> return MQTT
                }
            }
        }
    }

    enum class HttpAuthenticationType {
        NONE,
        BASIC;


        companion object {

            fun valueOf(ordinal: Int): HttpAuthenticationType {
                when (ordinal) {
                    0 -> return NONE
                    else -> return BASIC
                }
            }
        }
    }

    enum class MqttAuthenticationType {
        NONE,
        BASIC;


        companion object {

            fun valueOf(ordinal: Int): MqttAuthenticationType {
                when (ordinal) {
                    0 -> return NONE
                    else -> return BASIC
                }
            }
        }
    }

    enum class DataType {
        BOOLEAN,
        INTEGER,
        DECIMAL,
        STRING;


        companion object {

            fun valueOf(ordinal: Int): DataType {
                when (ordinal) {
                    0 -> return BOOLEAN
                    1 -> return INTEGER
                    2 -> return DECIMAL
                    else -> return STRING
                }
            }
        }
    }

    enum class MessageType {
        XML,
        JSON,
        RAW;


        companion object {

            fun valueOf(ordinal: Int): MessageType {
                when (ordinal) {
                    0 -> return XML
                    1 -> return JSON
                    else -> return RAW
                }
            }
        }
    }

    enum class MqttQosLevel {
        QOS_0,
        QOS_1,
        QOS_2;


        companion object {

            fun valueOf(ordinal: Int): MqttQosLevel {
                when (ordinal) {
                    0 -> return QOS_0
                    1 -> return QOS_1
                    else -> return QOS_2
                }
            }
        }
    }

    enum class HttpMethod {
        GET,
        PUT,
        POST,
        PATCH;

        override fun toString(): String {
            when (ordinal) {
                0 -> return "GET"
                1 -> return "PUT"
                2 -> return "POST"
                else -> return "PATCH"
            }
        }

        companion object {

            fun valueOf(ordinal: Int): HttpMethod {
                when (ordinal) {
                    0 -> return GET
                    1 -> return PUT
                    2 -> return POST
                    else -> return PATCH
                }
            }
        }
    }

    enum class LogLevel {
        INFO,
        WARN,
        ERROR,
        NOTIF_NONE,
        NOTIF_WARN,
        NOTIF_CRITICAL;

        override fun toString(): String {
            when (ordinal) {
                0 -> return "INFO"
                1 -> return "WARNING"
                2 -> return "ERROR"
                3 -> return "NOTIF_NONE"
                4 -> return "NOTIF_WARN"
                else -> return "NOTIF_CRITICAL"
            }
        }

        companion object {

            fun valueOf(ordinal: Int): LogLevel {
                when (ordinal) {
                    0 -> return INFO
                    1 -> return WARN
                    2 -> return ERROR
                    3 -> return NOTIF_NONE
                    4 -> return NOTIF_WARN
                    else -> return NOTIF_CRITICAL
                }
            }
        }
    }

    enum class NotificationType {
        NONE,
        WARNING,
        CRITICAL;

        override fun toString(): String {
            when (ordinal) {
                0 -> return "INFO"
                1 -> return "NOTIF_WARN"
                else -> return "NOTIF_CRITICAL"
            }
        }

        companion object {

            fun valueOf(ordinal: Int): NotificationType {
                when (ordinal) {
                    0 -> return NONE
                    1 -> return WARNING
                    else -> return CRITICAL
                }
            }
        }
    }

    enum class ElementManagementFunction {
        CREATE,
        UPDATE,
        DELETE;

        override fun toString(): String {
            when (ordinal) {
                0 -> return "CREATE"
                1 -> return "UPDATE"
                else -> return "DELETE"
            }
        }

        companion object {

            fun valueOf(ordinal: Int): ElementManagementFunction {
                when (ordinal) {
                    0 -> return CREATE
                    1 -> return UPDATE
                    else -> return DELETE
                }
            }
        }
    }

    enum class NotificationState {
        NONE,
        WARNING,
        CRITICAL;

        override fun toString(): String {
            when (ordinal) {
                0 -> return "NONE"
                1 -> return "WARNING"
                else -> return "CRITICAL"
            }
        }

        companion object {

            fun valueOf(ordinal: Int): NotificationState {
                when (ordinal) {
                    0 -> return NONE
                    1 -> return WARNING
                    else -> return CRITICAL
                }
            }
        }
    }

}
